### PR TITLE
Add a check / test which tries to uninstall system packages

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds a check which verifies that the system package uninstall works.

Related to #594 #595

### CODE REVIEW

File: rake/spec/default/99-uninstall-system-packages_spec.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Check that StackStorm packages can be uninstalled without errors
+describe 'st2 packages uninstall test' do
+  describe command("$BASEDIR/scripts/remove_os_packages.sh #{spec[:package_list]}") do
```
Comment (armab) [inline rake/spec/default/99-uninstall-system-packages_spec.rb]: Don't need a script.

Can check the OS flavor in Serverspec/Rspec
I could find https://serverspec.org/advanced_tips.html#how-to-get-os-information

Comment (Kami) [reply rake/spec/default/99-uninstall-system-packages_spec.rb]: Thanks - I was actually just looking into that :)

**Comment by Kami** [comment]
Looks like the tests are passing / working, minus the intermediate Circle CI issue / failure we are seeing...

Thanks to @nmaludy for bringing this up.

File: rake/spec/default/99-uninstall-system-packages_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      end
+
+      # Verify package has been uninstalled successfully
+      describe command("rpm -q #{pkg_name}") do
+        its(:exit_status) { is_expected.to eq 1 }
+      end
+    elsif os[:family] == 'ubuntu'
+      # Verify package is installed
+      describe command("dpkg -l #{pkg_name}") do
+        its(:exit_status) { is_expected.to eq 0 }
```
[the remark this answers is not in the document]
Comment (Kami) [reply rake/spec/default/99-uninstall-system-packages_spec.rb]: That's fair enough - I did that first but then I thought those additional checks don't hurt and may help :)

[the remark this answers is not in the document]
Comment (Kami) [reply rake/spec/default/99-uninstall-system-packages_spec.rb]: Having said that - I'm fine either way and I'm leaning towards more checks is better - those checks don't slow things down and additional checks are usually good.

**Comment by Kami** [comment]
@armab Removed more advanced checks and confirmed the test works as it shoudl by reverting a fix for packaging issue - https://circleci.com/gh/StackStorm/st2-packages/3258#tests/containers/3

**Pull request merged**